gormschema/gorm.go

```diff
@@ -187,3 +224,56 @@ func WithJoinTable(model any, field string, jointable any) Option {
 		})
 	}
 }
+
+func indirect(t reflect.Type) reflect.Type {
+	for t.Kind() == reflect.Ptr {
+		t = t.Elem()
+	}
+	return t
+}
+
+type (
+	// ViewOption is a configuration option for creating a viewBuilder.
+	ViewOption interface {
+		apply(*viewBuilder)
+	}
+	// ViewOptionFunc is an adapter to allow the use of functions as ViewOption.
+	ViewOptionFunc func(*viewBuilder)
+	viewDefiner    interface {
+		ViewDef() []ViewOption
+	}
+	viewBuilder struct {
+		db         *gorm.DB
+		createStmt string
+		// viewName is only used for the BuildStmt option.
+		// BuildStmt returns only a subquery; viewName helps to create a full CREATE VIEW statement.
+		viewName string
+	}
+)
+
+// CreateStmt accepts raw SQL with args to create a CREATE VIEW statement.
+func CreateStmt(sql string, args ...any) ViewOptionFunc {
+	return func(b *viewBuilder) {
+		b.createStmt = b.db.ToSQL(func(tx *gorm.DB) *gorm.DB {
+			return tx.Exec(sql, args...)
+		})
+	}
+}
+
+// BuildStmt accepts a function with gorm query builder to create a CREATE VIEW statement.
+// With this option, the view's name will be the same as the model's table name
+func BuildStmt(fn func(db *gorm.DB) *gorm.DB) ViewOptionFunc {
+	return func(b *viewBuilder) {
+		vd := b.db.ToSQL(func(tx *gorm.DB) *gorm.DB {
+			// Unscoped() helps to remove `where deleted_at is not null` from the query when model has gorm.Model embedded
+			// Find(nil) helps to execute the query within the context of ToSQL
+			return fn(tx).Unscoped().Find(nil)
```

Comment:
I don't think the user will be embed `gorm.Model` into structure for a view.

Reply:
Oh, sorry for the wrong information. The subquery looks something like this:

```
db.Model(User{})...
```

This subquery creates a `deleted_at` condition, there is no thing about embedded `gorm.Model`."